x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -369,8 +369,10 @@ export function addNewFeatureToIndex(geometry: Geometry | Position[]) {
     }
 
     try {
+      await dispatch(updateEditShape(DRAW_SHAPE.INSERT));
       await (layer as IVectorLayer).addFeature(geometry);
       await dispatch(syncDataForLayerDueToDrawing(layer));
+      await dispatch(updateEditShape(DRAW_SHAPE.SIMPLE_SELECT));
```

Comment:
The wait state is not being displayed while adding features. The problem is that `DrawFeatureControl._onDraw` calls this.props.disableDrawState, setting edit mode to `null` after `addNewFeatureToIndex` sets it to `WAIT`.

The solution will be to clean up DrawFeatureControl._onDraw to not call this.props.disableDrawState.